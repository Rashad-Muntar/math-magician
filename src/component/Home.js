import React from 'react';
import '../App.css';

const Home = () => (
  <div className="homeWrapper">
    <img className="homeImg" src="./download.jpeg" alt="home" />
    <p>WE LOVE MATH</p>
  </div>
);
export default Home;
