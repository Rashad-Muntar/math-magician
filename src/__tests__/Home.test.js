import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Home from '../component/Home';


describe('Home', () => {
  beforeEach(() => {
    render(<Home />);
  });

  it('Renders Quote component', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders Home correctly', () => {
    expect(screen.getByText(/WE LOVE MATH/)).toBeInTheDocument();
  });

  test('renders only the home page', () => {
    expect(screen.queryByText(/Quote of the day/)).not.toBeInTheDocument();
  });
});
