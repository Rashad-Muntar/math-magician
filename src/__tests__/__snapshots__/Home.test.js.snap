// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders Huote component 1`] = `
<div
  className="homeWrapper"
>
  <img
    alt="home"
    className="homeImg"
    src="./download.jpeg"
  />
  <p>
    WE LOVE MATH
  </p>
</div>
`;

exports[`Home Renders Quote component 1`] = `
<div
  className="homeWrapper"
>
  <img
    alt="home"
    className="homeImg"
    src="./download.jpeg"
  />
  <p>
    WE LOVE MATH
  </p>
</div>
`;
